File: 7_Kyu_Kata/sum-of-all-the-multiples-of-3-or-5/Swift/sumOfAllTheMultiplesOf3Or5.Test.swift
// [object Object] - Sum of all the multiples of 3 or 5  [ ID: 57f36495c0bb25ecf50000e7  (sum-of-all-the-multiples-of-3-or-5) ]
// URL: https://www.codewars.com/kata/57f36495c0bb25ecf50000e7
// Category: REFERENCE  |  Tags: FUNDAMENTALS
// *****************************************************************************
import XCTest

class SolutionTest: XCTestCase {
    static var allTests = [
        ("Basic Tests", basicTests),
        ("Random Tests", randomTests),
    ]

    func basicTests() {
        XCTAssertEqual(findSum(5), 8)
        XCTAssertEqual(findSum(10), 33)
        XCTAssertEqual(findSum(100), 2418)
        XCTAssertEqual(findSum(1000), 234168)
    }
    
    func randomTests() {
    
        func findSumdThI(_ n: Int) -> Int {
            var sum = 0
            for index in 0...n {
                if (index%3==0 || index%5==0) {
                    sum += index
                }
            }
            return sum
        }
    
        func rand(_ min: Int, _ max: Int) -> Int {
            return random() % (max - min) + min
        }
        for _ in 1...100 {
          let testN = rand(100, 100000)
          XCTAssertEqual(findSum(testN), findSumdThI(testN))
        }  
    }
}

XCTMain([
    testCase(SolutionTest.allTests),
])

File: 7_Kyu_Kata/sum-of-all-the-multiples-of-3-or-5/Python/sum_of_all_the_multiples_of_3_or_5_test.py
# [object Object] - Sum of all the multiples of 3 or 5  [ ID: 57f36495c0bb25ecf50000e7  (sum-of-all-the-multiples-of-3-or-5) ]
# URL: https://www.codewars.com/kata/57f36495c0bb25ecf50000e7
# Category: REFERENCE  |  Tags: FUNDAMENTALS
# ******************************************************************************
test.describe("Basic Tests")
test.it("Expected 5 to equal 8")
test.assert_equals(find(5), 8)
test.it("Expected 10 to equal 33")
test.assert_equals(find(10), 33)
test.it("Expected 100 to equal 2418")
test.assert_equals(find(100), 2418)
test.it("Expected 1000 to equal 234168")
test.assert_equals(find(1000), 234168)
import random
test.describe("Random Tests")
def findSOLVEIT(n):
    result = 0
    for i in range(3,n+1):
        if i%3==0 or i%5==0:
            result += i
    return result
for cwtests in range(0,96):
    num = random.randint(1,100000)
    result = findSOLVEIT(num)
    test.it("Expected "+str(num)+" to equal "+str(result))
    test.assert_equals(find(num), result)
